### PR TITLE
feat: Add supporting migrations to database

### PR DESCRIPTION
**Pull request by gagarinkomar**
Added supporting migrations using alembic library.
Added init migration.
Add param testing for DataBaseApi, because postgresql migrations don't want to work with sqlite.

TODO:
1)test corner cases
2)add tests scripts
3)try to fix problems with sqlite

### CODE REVIEW

File: manytask/migrations/env.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+load_dotenv("../.env")
+
+
+# this is the Alembic Config object, which provides
+# access to the values within the .ini file in use.
+config = context.config
+
+# config.set_main_option("DATABASE_URL", os.environ["DATABASE_URL"])
+# config.set_main_option("SCRIPT_LOCATION", str(Path(__file__).parent / 'migrations'))
```
Comment (prawwtocol) [inline manytask/migrations/env.py:17]: such comments should be removed if they're not needed, because if we merge them now they will remain here forever.

File: manytask/database.py [hunk 2]
```diff
@@ -39,13 +44,13 @@ def __init__(
...
         :param show_allscores: flag for showing results to all users
-        :param create_tables_if_not_exist: flag for creating database tables if they don't exist
+        :param apply_migrations: flag for applying migrations
         """
 
         self.engine = create_engine(database_url, echo=False)
 
-        if create_tables_if_not_exist:
-            self._create_tables()
+        if apply_migrations:
```
Comment (prawwtocol) [inline manytask/database.py:52]: I think it's better to do the following:

1. Always check for pending migrations
2. If apply_migrations=False, log a warning or raise an error

File: manytask/migrations/env.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import os
+
+from alembic import context
+from dotenv import load_dotenv
+from sqlalchemy import engine_from_config, pool
+
+from manytask.models import Base
+
+load_dotenv("../.env")
```
Comment (prawwtocol) [inline manytask/migrations/env.py:9]: I don't like this relative path. 


I like the following: `Path(__file__).parent.parent.parent / '.env'` but just a tiny bit better.

Why do we need to load_dotenv?

File: Makefile [hunk 4]
```diff
@@ -43,4 +43,11 @@ format:
...
-	mypy --install-types --non-interactive $(ROOT_DIR) $(TESTS_DIR)
\ No newline at end of file
+	mypy --install-types --non-interactive $(ROOT_DIR) $(TESTS_DIR)
+
+makemigrations:
+	@ if [ -z "$(msg)" ]; then \
+		echo "Environment variable msg not set"; \
+		exit 1; \
+	fi
+	docker-compose -f $(DOCKER_COMPOSE_DEV) run -e EMPTY_APP=true --rm -w /app/manytask/ manytask python -m alembic revision -m "$(msg)" --autogenerate
```
Comment (prawwtocol) [inline Makefile:53]: don't we want to simplify this command? require make dev and then just run alembic upgrade head or something similar?

**Review by prawwtocol — COMMENTED** [review]
~I think we need to update init-db.sh to run migrations~

**Review by prawwtocol — COMMENTED** [review]
Also, don't we want to exclude the migrations directory from our linitng rules?

File: manytask/alembic.ini [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+# set to 'true' to search source files recursively
+# in each "version_locations" directory
+# new in Alembic version 1.10
+# recursive_version_locations = false
+
+# the output encoding used when revision files
+# are written from script.py.mako
+# output_encoding = utf-8
+
+# sqlalchemy.url = %(DATABASE_URL)s
```
Comment (prawwtocol) [inline manytask/alembic.ini:66]: please add a comment on why this is commented out

File: Makefile (hunk 4 above)
Comment (akostrikov) [reply Makefile:53]: Also, we need to be able to migrate code. Like `make migrate`, `make downgrade`

File: manytask/database.py (hunk 2 above)
Comment (akostrikov) [reply manytask/database.py:52]: Lets log error level for now. Maybe raise error later

**Review by akostrikov — CHANGES_REQUESTED** [review]

File: manytask/migrations/env.py (hunk 3 above)
Comment (gagarinkomar) [reply manytask/migrations/env.py]: Changed to dotenv_values and fixed path. We need this to import database_url from .env.

**Review by akostrikov — APPROVED** [review]
LGTM, lets ask @prawwtocol

**Review by akostrikov — APPROVED** [review]

**Comment by prawwtocol** [comment]
LGTM!
